### ISSUE TITLE
Wrong UV planes when converting from Y32

### ISSUE THREAD

**Issue by magiblot** [pinterf/AviSynthPlus#18]
Many conversion functions give weird results with 32-bit Y sources, most probably due to not being adapted to the new range for the U and V channels. It looks like they want to fill the U and V planes with `half_range` but they fill them with `0.5` instead of `0.0`.

Y8 source clip:
![imatge](https://user-images.githubusercontent.com/20713561/43670009-e542746e-9787-11e8-80fc-fba055f831fb.png)

```
ConvertBits(32)
ConvertToYV12() #or YV16, or YV24
ConvertBits(8)
```

![imatge](https://user-images.githubusercontent.com/20713561/43670020-00bf9a6e-9788-11e8-84e4-e3b007a57ea7.png)

```
ConvertBits(32)
ConvertToYV12() #or YV16, or YV24
Expr("x","x 0.5 -")
ConvertBits(8)
```
![imatge](https://user-images.githubusercontent.com/20713561/43670030-17d7bf10-9788-11e8-814a-a2fe65b7b9be.png)

```
ConvertBits(16)
ConvertToYV12() #or YV16, or YV24
ConvertBits(8)
```
![imatge](https://user-images.githubusercontent.com/20713561/43670061-5702606e-9788-11e8-8a07-78091b4902fd.png)

In addition, converting Y to YUY2 at any bitrate other than 8 will cause a severe crash:

![imatge](https://user-images.githubusercontent.com/20713561/43670329-cf14d36e-978a-11e8-8158-81860029f2a6.png)

I'm sorry.

**Comment by magiblot** [comment]
Thank you very much!

Yeah, I had forgotten `YUY2` is only intended for a bit depth of 8.

However, other Convert functions (`YV12`, `YV16`, `YV24`) keep the input bit depth, which is a bit confusing, considering the tables in http://avisynth.nl/index.php/Convert, although this may be the intended behaviour.

At first I thought they would downsample to output to 8 bit, but they seem to be synonims for `ConvertToYUV420`, `YUV422` and `YUV444`, respectively.

**Comment by pinterf** [comment]
Yes, I have recognized that behaviour (ConvertToYV12..24 keeping the bitdepth), but forgot about it. It is not intended, perhaps it was introduced in an early internal reorganization. Maybe I'll look at it but it is not priority.

**Comment by magiblot** [comment]
Alright. Thanks!

**Comment by magiblot** [comment]
By the way, I have noticed when converting from RGB to YUV that `RGB->YV24->YV12` is much faster than `RGB->YV12`. Why is that? My processor supports AVX at most.

**Comment by pinterf** [comment]
RGB is always converted to YV24 internally before doing YV12 conversion. So there is no reason to get noticable difference. You may start a different ticket on that and I'll check it.

**Comment by magiblot** [comment]
Okay. I'll do that.
